src/main/java/training/supportbank/DataHandler.java

```diff
@@ -0,0 +1,77 @@
+package training.supportbank;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+public class DataHandler {
+    public static List<Map<String, String>> dataConverter(BufferedReader reader) throws IOException {
+        String transactionLine;
+        List<Map<String, String>> transactionArray = new ArrayList<>();
+
+        while ((transactionLine = reader.readLine()) != null) {
+            Map<String, String> transactionMap = new LinkedHashMap<>();
```

Comment:
Using a map to represent a transaction works, but it has a few downsides:

- The other parts of your code need to know what the keys of the map are
- Because the map's objects are all `String`s, you're forced to represent everything (including dates and money) as strings
- Your code passes around `Map<String, String>` a lot. You know what that means because you wrote the code, but it wouldn't be immediately obvious to a developer who's new to the codebase.

How else could you represent the transaction to solve these problems? Give me a shout if you want to talk it through. 🙂

Reply:
Would using `List<Object> = new ArrayList<>()` be better? My thought-process here is that I could create an Array of Arrays (instead of an Array of Maps) whereby the inner Arrays will always have the same structure. This way, we can use multiple object types and we know which index to access for the relevant item (date, from, to, narrative or amount).

Reply:
Now created a new Transaction class